p4-16/psa/PSA.mdk

```diff
@@ -180,9 +180,12 @@ the contents of several metadata fields in the struct
     if (ostd.multicast_group != 0) {
         Make 0 or more copies of the packet according to the control
             plane configuration of multicast group ostd.multicast_group.
+            Every copy will have the same value of ostd.class_of_service
         return;   // Do not continue below.
     }
-    enqueue one packet for output port ostd.egress_port
+    enqueue one packet for output port ostd.egress_port with class of
+        service specified by ostd.class_of_service
```

Comment:
I think it might be useful to add what should happen if a given class of service doesn't exist on a particular target. In other words, I have a program that send packet with class-of-service 7 and try to run this program on a system which has only one queue....

Option 1: If class-of-service doesn't exist, drop the packet
Option 2: If class-of-service doesn't exist, use one of the existing classes as per platform's choice.

Reply:
I added an 'if' statement to the pseudocode describing what happens to the packet at the end of ingress processing, that specifies option 1, with optional error status recorded about the drop.

Reply:
From 2017-Sep-07 PSA working group meeting, updated the behavior in the case where the user program assigns a value to class_of_service that is not supported by the platform, to use class_of_service 0, instead of dropping the packet.